src/structurizr/view/view_set.py

```diff
@@ -279,16 +291,50 @@ def create_dynamic_view(self, **kwargs) -> DynamicView:
         Args:
             **kwagrs: Provide keyword arguments for instantiating a `DynamicView`.
         """
-        # TODO:
-        # AssertThatTheViewKeyIsUnique(key);
         dynamic_view = DynamicView(**kwargs)
+        self._ensure_key_is_specific_and_unique(dynamic_view.key)
         dynamic_view.set_viewset(self)
         dynamic_view.set_model(self.model)
         self.dynamic_views.add(dynamic_view)
         return dynamic_view
 
+    def create_filtered_view(self, **kwargs) -> FilteredView:
+        """
+        Add a new FilteredView to the ViewSet.
+
+        Args:
+            **kwargs: Provide keyword arguments for instantiating a `FilteredView`.
+        """
+        filtered_view = FilteredView(**kwargs)
+        self._ensure_key_is_specific_and_unique(filtered_view.key)
+        filtered_view.set_viewset(self)
+        self.filtered_views.add(filtered_view)
+        return filtered_view
+
+    def get_view(self, key: str) -> Optional[AbstractView]:
+        """Return the view with the given key, or None."""
+        all_views = (
+            self.system_landscape_views
+            | self.system_context_views
+            | self.container_views
+            | self.component_views
+            | self.deployment_views
+            | self.dynamic_views
+            | self.filtered_views
+        )
+        return next((view for view in all_views if view.key == key), None)
+
+    def __getitem__(self, key: str) -> AbstractView:
```

Comment:
This method points a bit towards creating a 'private' attribute which is a `dict` of all the views. Then fetching a particular view by key is very easy. It would mean managing that dictionary, though.

Reply:
OK, so I've had a crack at this - actually replacing the various sets with a single dictionary.  I think this actually works quite well - see what you think.